pkg/jobrunaggregator/jobrunaggregatorlib/jobrun_locator_per_pr_jobs.go

```diff
@@ -1,19 +1,30 @@
 package jobrunaggregatorlib
 
 import (
-	"fmt"
 	"time"
 
+	"github.com/sirupsen/logrus"
+
 	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
 )
 
 const (
-	// AggregationIDLabel is the name of the label for the aggregation id in prow job
-	AggregationIDLabel = "release.openshift.io/aggregation-id"
-	// PayloadInvocationIDLabel is the name of the label for the payload invocation id in prow job
-	PayloadInvocationIDLabel = "release.openshift.io/aggregation-id"
+	// ProwJobAggregationIDLabel is the name of the label for the aggregation id in prow job
+	ProwJobAggregationIDLabel = "release.openshift.io/aggregation-id"
+	// ProwJobPayloadInvocationIDLabel is the name of the label for the payload invocation id in prow job
+	ProwJobPayloadInvocationIDLabel = "release.openshift.io/aggregation-id"
+	// prowJobReleaseJobNameAnnotation refers to the original periodic job name for PR based payload runs
+	prowJobReleaseJobNameAnnotation = "releaseJobName"
 )
 
+func NewProwJobMatcherFuncForPR(jobName, matchID, matchLabel string) ProwJobMatcherFunc {
+	return perPRProwJobMatcher{
```

Comment:
Could this just be a closure?

Reply:
@stevekuznetsov Can you elaborate on this? Thanks!

Reply:
Right now you have:

```go
type perPRProwJobMatcher struct {
	// matchID is how we recognize per-PR payload jobs. It is set based on the matchLabel value
	// that the per-PR payload controller sets in the prowjobs it creates.
	matchID    string
	matchLabel string
}

func (a perPRProwJobMatcher) shouldAggregateReleaseControllerJob(prowJob *prowjobv1.ProwJob) bool {
	id := prowJob.Labels[a.matchLabel]
	jobName := prowJob.Labels["prow.k8s.io/job"]
	jobRunId := prowJob.Labels["prow.k8s.io/build-id"]
	fmt.Printf("  checking %v/%v for matchID match: looking for %q found %q.\n", jobName, jobRunId, a.matchID, id)
	idMatches := len(a.matchID) > 0 && id == a.matchID

	return idMatches
}
```


You could just as well have:

```go
func shouldAggregateReleaseControllerJob(matchId, matchLabel string) func(prowJob *prowjobv1.ProwJob) bool {
     return func(prowJob *prowjobv1.ProwJob) bool {
		id := prowJob.Labels[a.matchLabel]
		jobName := prowJob.Labels["prow.k8s.io/job"]
		jobRunId := prowJob.Labels["prow.k8s.io/build-id"]
		fmt.Printf("  checking %v/%v for matchID match: looking for %q found %q.\n", jobName, jobRunId, a.matchID, id)
		idMatches := len(a.matchID) > 0 && id == a.matchID

		return idMatches
	}
}
```

What is the point of the `perPRProwJobMatcher` type?